src/ObjLoading/ObjContainer/SoundBank/SoundBankWriter.cpp

```diff
@@ -0,0 +1,285 @@
+#include "SoundBankWriter.h"
+
+#include "Crypto.h"
+#include "ObjContainer/SoundBank/SoundBankTypes.h"
+#include "Sound/WavTypes.h"
+#include "Utils/Alignment.h"
+#include "Utils/FileUtils.h"
+
+#include <cstring>
+#include <filesystem>
+#include <iostream>
+#include <unordered_map>
+
+std::unordered_map<unsigned int, unsigned char> INDEX_FOR_FRAMERATE{
+    {8000,   0},
+    {12000,  1},
+    {16000,  2},
+    {24000,  3},
+    {32000,  4},
+    {44100,  5},
+    {48000,  6},
+    {96000,  7},
+    {192000, 8},
+};
+
+class SoundBankWriterImpl : public SoundBankWriter
+{
+    static constexpr char BRANDING[] = "Created with OAT - OpenAssetTools";
+    static constexpr int64_t DATA_OFFSET = 0x800;
+    static constexpr uint32_t MAGIC = FileUtils::MakeMagic32('2', 'U', 'X', '#');
+    static constexpr uint32_t VERSION = 14u;
+
+    inline static const std::string PAD_DATA = std::string(16, '\x00');
+
+public:
+    explicit SoundBankWriterImpl(const std::string& fileName, std::ostream& stream, ISearchPath* assetSearchPath)
+        : m_file_name(fileName),
+          m_stream(stream),
+          m_asset_search_path(assetSearchPath),
+          m_sounds(),
+          m_current_offset(0),
+          m_total_size(0),
+          m_entry_section_offset(0),
+          m_checksum_section_offset(0)
+    {
+    }
+
+    void AddSound(const std::string& soundFilePath, unsigned int soundId, bool looping, bool streamed) override
+    {
+        this->m_sounds.push_back({soundFilePath, soundId, looping, streamed});
+    }
+
+    void GoTo(const int64_t offset)
+    {
+        m_stream.seekp(offset, std::ios::beg);
+        m_current_offset = offset;
+    }
+
+    void Write(const void* data, const size_t dataSize)
+    {
+        m_stream.write(static_cast<const char*>(data), dataSize);
+        m_current_offset += dataSize;
+    }
+
+    void Pad(const size_t paddingSize)
+    {
+        auto paddingSizeLeft = paddingSize;
+        while (paddingSizeLeft > 0)
+        {
+            const auto writeSize = std::min(paddingSizeLeft, PAD_DATA.size());
+            Write(PAD_DATA.data(), writeSize);
+
+            paddingSizeLeft -= writeSize;
+        }
+    }
+
+    void AlignToChunk()
+    {
+        if ((m_current_offset & 0xF) != 0)
+            Pad(0x10 - (m_current_offset & 0xF));
```

Comment:
```cpp
if (m_current_offset % 16 != 0)
    Pad(m_current_offset % 16);
```
I think that's more readable 😅

Reply:
I changed it to use mod, but the example you gave does not do the same thing without the "0x10 - " in the pad call

Reply:
yeah you're right, messed up the code suggestion 👍